ha/setup/create_pacemaker_resources.py

```diff
@@ -0,0 +1,251 @@
+    cmd_clone_motr = f"pcs -f {cib_xml} resource clone motr"
+
+    for s in (cmd_hax, cmd_confd, cmd_ios, cmd_clone_motr):
+        SimpleCommand().run_cmd(s)
+
+    if push:
+        cib_push(cib_xml)
+
+
+def free_space_monitor(cib_xml, push=False):
+    """Create free space monitor resource. 1 per cluster, no affinity."""
+    cmd_fsm = f"pcs -f {cib_xml} resource create motr-free-space-mon systemd:motr-free-space-monitor op monitor interval=30s meta failure-timeout=300s"
+    SimpleCommand().run_cmd(cmd_fsm)
+
+    constraints = [
+            f"pcs -f {cib_xml} constraint order motr-ios then motr-free-space-mon",
+            f"pcs -f {cib_xml} constraint colocation add motr-free-space-mon with motr-ios"
+            ]
+    for c in constraints:
+        SimpleCommand().run_cmd(c)
+
+    if push:
+        cib_push(cib_xml)
+
+
+def s3servers(cib_xml, push=False):
+    """Create resources that belong to s3server group and clone the group.
+
+    S3 background consumer is ordered after s3server and co-located with it."""
+    for i in range(1, 12):
+        cmd_s3server = f"pcs -f {cib_xml} resource create s3server-{i} ocf:seagate:dynamic_fid_service_ra service=s3server fid_service_name=s3service --group s3server"
+        SimpleCommand().run_cmd(cmd_s3server)
+
+    cmd_s3bc = f"pcs -f {cib_xml} resource create s3backcons systemd:s3backgroundconsumer meta failure-timeout=300s --group s3server"
+    SimpleCommand().run_cmd(cmd_s3bc)
+
+    cmd_clone_s3server = f"pcs -f {cib_xml} resource clone s3server"
+    SimpleCommand().run_cmd(cmd_clone_s3server)
+
+    if push:
+        cib_push(cib_xml)
+
+
+def s3bp(cib_xml, push=False):
+    """Create S3 background producer
+
+    S3 background producer have to be only 1 per cluster and co-located with
+    s3server."""
+    cmd_s3bp = f"pcs -f {cib_xml} resource create s3backprod systemd:s3backgroundproducer op monitor interval=30s"
+    cmd_order = f"pcs -f {cib_xml} constraint order s3server-clone then s3backprod"
+
+    for s in (cmd_s3bp, cmd_order):
+        SimpleCommand().run_cmd(s)
+
+    if push:
+        cib_push(cib_xml)
+
+
+def s3auth(cib_xml, push=False):
+    """Create haproxy S3 auth server resource in pacemaker."""
+    cmd_s3auth = f"pcs -f {cib_xml} resource create s3auth systemd:s3authserver clone op monitor interval=30"
+    SimpleCommand().run_cmd(cmd_s3auth)
+
+    if push:
+        cib_push(cib_xml)
+
+
+def haproxy(cib_xml, push=False):
+    """Create haproxy clone resource in pacemaker."""
+    cmd_haproxy = f"pcs -f {cib_xml} resource create haproxy systemd:haproxy clone op monitor interval=30"
+    SimpleCommand().run_cmd(cmd_haproxy)
+
+    if push:
+        cib_push(cib_xml)
+
+
+def sspl(cib_xml, push=False):
+    """Create sspl clone resource in pacemaker."""
+    # Using sspl-ll service file according to the content of SSPL repo
+    cmd_sspl = f"pcs -f {cib_xml} resource create sspl-ll systemd:sspl-ll clone op monitor interval=30"
```

Comment:
sspl should be part of monitor group. This group is going to have single resource for now.

Reply:
Done